### ISSUE TITLE
problem not found lexer in Arch linux-lts

### ISSUE THREAD

**Issue by MOHAMED19OS** [crosstool-ng/crosstool-ng#1790]
### Issue submission guidelines
See [detaled guidelines](http://crosstool-ng.github.io/support/). In short:
- the least version in GitHub
- Arch Linux x86_64 , kernel 5.15.55-1-lts ,  Kde Plasma 5.25.3

download  :
`git clone --depth 1 https://github.com/crosstool-ng/crosstool-ng.git ; cd crosstool-ng && ./bootstrap && ./configure --enable-local && make
`

bootstrap:

INFO  :: *** Generating package version descriptions
INFO  :: Master packages: autoconf automake avr-libc binutils bison cloog dtc duma elf2flt expat gcc gdb gettext glibc gmp gnuprumcu isl libelf libiconv libtool linux ltrace m4 make mingw-w64 moxiebox mpc mpfr musl ncurses newlib-nano newlib picolibc strace uClibc-ng zlib
INFO  :: Generating 'config/versions/autoconf.in'
INFO  :: Generating 'config/versions/automake.in'
INFO  :: Generating 'config/versions/avr-libc.in'
INFO  :: Generating 'config/versions/binutils.in'
INFO  :: Generating 'config/versions/bison.in'
INFO  :: Generating 'config/versions/cloog.in'
INFO  :: Generating 'config/versions/dtc.in'
INFO  :: Generating 'config/versions/duma.in'
INFO  :: Generating 'config/versions/elf2flt.in'
INFO  :: Generating 'config/versions/expat.in'
INFO  :: Generating 'config/versions/gcc.in'
INFO  :: Generating 'config/versions/gdb.in'
INFO  :: Generating 'config/versions/gettext.in'
INFO  :: Generating 'config/versions/glibc.in'
INFO  :: Generating 'config/versions/gmp.in'
INFO  :: Generating 'config/versions/gnuprumcu.in'
INFO  :: Generating 'config/versions/isl.in'
INFO  :: Generating 'config/versions/libelf.in'
INFO  :: Generating 'config/versions/libiconv.in'
INFO  :: Generating 'config/versions/libtool.in'
INFO  :: Generating 'config/versions/linux.in'
INFO  :: Generating 'config/versions/ltrace.in'
INFO  :: Generating 'config/versions/m4.in'
INFO  :: Generating 'config/versions/make.in'
INFO  :: Generating 'config/versions/mingw-w64.in'
INFO  :: Generating 'config/versions/moxiebox.in'
INFO  :: Generating 'config/versions/mpc.in'
INFO  :: Generating 'config/versions/mpfr.in'
INFO  :: Generating 'config/versions/musl.in'
INFO  :: Generating 'config/versions/ncurses.in'
INFO  :: Generating 'config/versions/newlib-nano.in'
INFO  :: Generating 'config/versions/newlib.in'
INFO  :: Generating 'config/versions/picolibc.in'
INFO  :: Generating 'config/versions/strace.in'
INFO  :: Generating 'config/versions/uClibc-ng.in'
INFO  :: Generating 'config/versions/zlib.in'
INFO  :: *** Generating menu/choice selections
INFO  :: Generating arch.in (choice)
INFO  :: Generating kernel.in (choice)
INFO  :: Generating cc.in (choice)
INFO  :: Generating binutils.in (choice)
INFO  :: Generating libc.in (choice)
INFO  :: Generating debug.in (menu)
INFO  :: Generating comp_tools.in (menu)
INFO  :: Generating comp_libs.in (menu)
INFO  :: *** Gathering the list of data files to install
INFO  :: *** Running autoreconf
INFO  :: *** Done!

the error  in make :

/bin/sh ./config.status --recheck
running CONFIG_SHELL=/bin/sh /bin/sh ./configure --enable-local --no-create --no-recursion
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether ln -s works... yes
checking whether install takes --strip-program option... yes
checking for ggrep... no
checking for grep... grep
checking for absolute path to grep... /usr/bin/grep
checking for gegrep... no
checking for egrep... egrep
checking for absolute path to egrep... /usr/bin/egrep
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for sed... /usr/bin/sed
checking for GNU sed >= 4.0... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking how to run the C preprocessor... gcc -E
checking for flex... flex
checking for lex output file root... lex.yy
checking for lex library... none needed
checking for library containing yywrap... no
configure: WARNING: yywrap not found; giving up on flex
checking for bison... bison -y
checking whether g++ supports C++11 features with -std=gnu++11... yes
checking if gcc can static link... yes
checking for ranlib... ranlib
checking for gobjcopy... no
checking for objcopy... objcopy
checking for absolute path to objcopy... /usr/bin/objcopy
checking for gobjdump... no
checking for objdump... objdump
checking for absolute path to objdump... /usr/bin/objdump
checking for greadelf... no
checking for readelf... readelf
checking for absolute path to readelf... /usr/bin/readelf
checking for flex... flex
checking for makeinfo... makeinfo
checking for cut... cut
checking for readlink... readlink
checking for tar... tar
checking for gzip... gzip
checking for bzip2... bzip2
checking for xz... xz
checking for unzip... unzip
checking for help2man... help2man
checking for file... file
checking for which... which
checking for lzip... lzip
checking for wget... wget
checking for curl... curl
checking for meson... no
checking for ninja... no
checking for rsync... no
checking whether to use getconf or sysctl to count CPUs... getconf _NPROCESSORS_ONLN
checking for gpatch... no
checking for patch... patch
checking for absolute path to patch... /usr/bin/patch
checking whether patch supports --no-backup-if-mismatch... yes
checking for bash... /usr/bin/bash
checking for GNU bash >= 3.1... yes
checking for absolute path to gawk... /usr/bin/gawk
checking for GNU awk... yes
checking for make... /usr/bin/make
checking for GNU make >= 3.81... yes
checking for GNU make >= 4.0... yes
checking for libtool... /usr/bin/libtool
checking for GNU libtool >= 2.4... yes
checking for libtoolize... /usr/bin/libtoolize
checking for GNU libtoolize >= 2.4... yes
checking for GNU autoconf >= 2.65... yes
checking for autoreconf... /usr/bin/autoreconf
checking for GNU autoreconf >= 2.63... yes
checking for GNU automake >= 1.15... yes
checking for m4... /usr/bin/m4
checking for GNU m4 >= 1.4.12... yes
checking for python... /usr/bin/python3
checking for Python... yes
checking for Python >= 3.4... yes
checking for bison... /usr/bin/bison
checking for bison >= 2.7... yes
checking for dtc... no
checking for cvs... cvs
checking for svn... svn
checking for git... git
checking for gmd5sum... no
checking for md5sum... /usr/bin/md5sum
checking for gsha1sum... no
checking for sha1sum... /usr/bin/sha1sum
checking for gsha256sum... no
checking for sha256sum... /usr/bin/sha256sum
checking for gsha512sum... no
checking for sha512sum... /usr/bin/sha512sum
checking for inline... inline
checking for GNU libc compatible malloc... yes
checking for GNU libc compatible realloc... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking whether NLS is requested... yes
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for ncursesw via pkg-config... yes
checking for working ncursesw/curses.h... no
checking for working ncursesw.h... no
checking for working ncurses.h... yes
checking for Curses Panel library with ncursesw/panel.h... no
checking for Curses Panel library with panel.h... yes
checking for Curses Menu library with ncursesw/menu.h... no
checking for Curses Menu library with menu.h... yes
checking for build time... Mon Jul 25 11:46:49 2022
checking if the manual needs to be installed... no
checking that generated files are newer than configure... done
configure: creating ./config.status
 /bin/sh ./config.status
config.status: creating Makefile
config.status: creating paths.sh
config.status: creating kconfig/Makefile
config.status: creating config/configure.in
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
/usr/bin/make  all-recursive
make[1]: Entering directory '/home/mohamed/crosstool-ng'
Making all in kconfig
make[2]: Entering directory '/home/mohamed/crosstool-ng/kconfig'
  LEX      lexer.lex.c
/usr/bin/make  all-am
make[3]: Entering directory '/home/mohamed/crosstool-ng/kconfig'
  LEX      lexer.lex.c
  CC       lexer.lex.o
cc1: fatal error: lexer.lex.c: No such file or directory
compilation terminated.
make[3]: *** [Makefile:620: lexer.lex.o] Error 1
make[3]: Leaving directory '/home/mohamed/crosstool-ng/kconfig'
make[2]: *** [Makefile:468: all] Error 2
make[2]: Leaving directory '/home/mohamed/crosstool-ng/kconfig'
make[1]: *** [Makefile:2373: all-recursive] Error 1
make[1]: Leaving directory '/home/mohamed/crosstool-ng'
make: *** [Makefile:2120: all] Error 2

**Comment by cpackham** [comment]
hmm something odd has happened detecting flex

```
checking for flex... flex
checking for lex output file root... lex.yy
checking for lex library... none needed
checking for library containing yywrap... no
configure: WARNING: yywrap not found; giving up on flex
...
checking for flex... flex
```

And then when we try to create lexer.lex.c things blow up. I suspect the answer will be something along the lines of "install flex" but I'll try an reproduce the problem in an arch container first

**Comment by MOHAMED19OS** [comment]
@cpackham 

look i' test version Released 1.25.0 from 
https://crosstool-ng.github.io/2022/05/07/release-1.25.0.html
no problem

`wget http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.25.0.tar.xz ; tar xf crosstool-ng-1.25.0.tar.xz ; cd crosstool-ng-1.25.0 ; ./bootstrap && ./configure --enable-local && make
`




NFO  :: *** Generating package version descriptions
INFO  :: Master packages: android-ndk autoconf automake avr-libc binutils bison cloog dtc duma elf2flt expat gcc gdb gettext glibc glibc-ports gmp gnuprumcu isl libelf libiconv libtool linux ltrace m4 make mingw-w64 moxiebox mpc mpfr musl ncurses newlib-nano newlib picolibc strace uClibc-ng zlib
INFO  :: Generating 'config/versions/android-ndk.in'
INFO  :: Generating 'config/versions/autoconf.in'
INFO  :: Generating 'config/versions/automake.in'
INFO  :: Generating 'config/versions/avr-libc.in'
INFO  :: Generating 'config/versions/binutils.in'
INFO  :: Generating 'config/versions/bison.in'
INFO  :: Generating 'config/versions/cloog.in'
INFO  :: Generating 'config/versions/dtc.in'
INFO  :: Generating 'config/versions/duma.in'
INFO  :: Generating 'config/versions/elf2flt.in'
INFO  :: Generating 'config/versions/expat.in'
INFO  :: Generating 'config/versions/gcc.in'
INFO  :: Generating 'config/versions/gdb.in'
INFO  :: Generating 'config/versions/gettext.in'
INFO  :: Generating 'config/versions/glibc.in'
INFO  :: Generating 'config/versions/glibc-ports.in'
INFO  :: Generating 'config/versions/gmp.in'
INFO  :: Generating 'config/versions/gnuprumcu.in'
INFO  :: Generating 'config/versions/isl.in'
INFO  :: Generating 'config/versions/libelf.in'
INFO  :: Generating 'config/versions/libiconv.in'
INFO  :: Generating 'config/versions/libtool.in'
INFO  :: Generating 'config/versions/linux.in'
INFO  :: Generating 'config/versions/ltrace.in'
INFO  :: Generating 'config/versions/m4.in'
INFO  :: Generating 'config/versions/make.in'
INFO  :: Generating 'config/versions/mingw-w64.in'
INFO  :: Generating 'config/versions/moxiebox.in'
INFO  :: Generating 'config/versions/mpc.in'
INFO  :: Generating 'config/versions/mpfr.in'
INFO  :: Generating 'config/versions/musl.in'
INFO  :: Generating 'config/versions/ncurses.in'
INFO  :: Generating 'config/versions/newlib-nano.in'
INFO  :: Generating 'config/versions/newlib.in'
INFO  :: Generating 'config/versions/picolibc.in'
INFO  :: Generating 'config/versions/strace.in'
INFO  :: Generating 'config/versions/uClibc-ng.in'
INFO  :: Generating 'config/versions/zlib.in'
INFO  :: *** Generating menu/choice selections
INFO  :: Generating arch.in (choice)
INFO  :: Generating kernel.in (choice)
INFO  :: Generating cc.in (choice)
INFO  :: Generating binutils.in (choice)
INFO  :: Generating libc.in (choice)
INFO  :: Generating debug.in (menu)
INFO  :: Generating comp_tools.in (menu)
INFO  :: Generating comp_libs.in (menu)
INFO  :: *** Gathering the list of data files to install
INFO  :: *** Running autoreconf
configure.ac:96: warning: AC_PROG_LEX without either yywrap or noyywrap is obsolete
./lib/autoconf/programs.m4:716: _AC_PROG_LEX is expanded from...
./lib/autoconf/programs.m4:709: AC_PROG_LEX is expanded from...
configure.ac:96: the top level
configure.ac:303: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:303: You should run autoupdate.
./lib/autoconf/headers.m4:704: AC_HEADER_STDC is expanded from...
configure.ac:303: the top level
INFO  :: *** Done!
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether ln -s works... yes
checking whether install takes --strip-program option... yes
checking for ggrep... no
checking for grep... grep
checking for absolute path to grep... /usr/bin/grep
checking for gegrep... no
checking for egrep... egrep
checking for absolute path to egrep... /usr/bin/egrep
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for sed... /usr/bin/sed
checking for GNU sed >= 4.0... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking how to run the C preprocessor... gcc -E
checking for flex... flex
checking for lex output file root... lex.yy
checking for lex library... none needed
checking for library containing yywrap... no
checking whether yytext is a pointer... yes
checking for bison... bison -y
checking whether g++ supports C++11 features with -std=gnu++11... yes
checking if gcc can static link... yes
checking for ranlib... ranlib
checking for gobjcopy... no
checking for objcopy... objcopy
checking for absolute path to objcopy... /usr/bin/objcopy
checking for gobjdump... no
checking for objdump... objdump
checking for absolute path to objdump... /usr/bin/objdump
checking for greadelf... no
checking for readelf... readelf
checking for absolute path to readelf... /usr/bin/readelf
checking for flex... flex
checking for makeinfo... makeinfo
checking for cut... cut
checking for readlink... readlink
checking for tar... tar
checking for gzip... gzip
checking for bzip2... bzip2
checking for xz... xz
checking for unzip... unzip
checking for help2man... help2man
checking for file... file
checking for which... which
checking for lzip... lzip
checking for wget... wget
checking for curl... curl
checking for meson... no
checking for ninja... no
checking for rsync... no
checking whether to use getconf or sysctl to count CPUs... getconf _NPROCESSORS_ONLN
checking for gpatch... no
checking for patch... patch
checking for absolute path to patch... /usr/bin/patch
checking whether patch supports --no-backup-if-mismatch... yes
checking for bash... /usr/bin/bash
checking for GNU bash >= 3.1... yes
checking for absolute path to gawk... /usr/bin/gawk
checking for GNU awk... yes
checking for make... /usr/bin/make
checking for GNU make >= 3.81... yes
checking for GNU make >= 4.0... yes
checking for libtool... /usr/bin/libtool
checking for GNU libtool >= 2.4... yes
checking for libtoolize... /usr/bin/libtoolize
checking for GNU libtoolize >= 2.4... yes
checking for GNU autoconf >= 2.65... yes
checking for autoreconf... /usr/bin/autoreconf
checking for GNU autoreconf >= 2.63... yes
checking for GNU automake >= 1.15... yes
checking for m4... /usr/bin/m4
checking for GNU m4 >= 1.4.12... yes
checking for python... /usr/bin/python3
checking for Python... yes
checking for Python >= 3.4... yes
checking for bison... /usr/bin/bison
checking for bison >= 2.7... yes
checking for dtc... no
checking for cvs... cvs
checking for svn... svn
checking for git... git
checking for gmd5sum... no
checking for md5sum... /usr/bin/md5sum
checking for gsha1sum... no
checking for sha1sum... /usr/bin/sha1sum
checking for gsha256sum... no
checking for sha256sum... /usr/bin/sha256sum
checking for gsha512sum... no
checking for sha512sum... /usr/bin/sha512sum
checking for inline... inline
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for egrep... (cached) /usr/bin/grep -E
checking for GNU libc compatible malloc... yes
checking for GNU libc compatible realloc... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking whether NLS is requested... yes
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for ncursesw via pkg-config... yes
checking for working ncursesw/curses.h... no
checking for working ncursesw.h... no
checking for working ncurses.h... yes
checking for Curses Panel library with ncursesw/panel.h... no
checking for Curses Panel library with panel.h... yes
checking for Curses Menu library with ncursesw/menu.h... no
checking for Curses Menu library with menu.h... yes
checking for build time... Tue Jul 26 09:46:38 2022
checking if the manual needs to be installed... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating paths.sh
config.status: creating kconfig/Makefile
config.status: creating config/configure.in
config.status: creating config.h
config.status: executing depfiles commands
/usr/bin/make  all-recursive
make[1]: Entering directory '/home/mohamed/crosstool-ng-1.25.0'
Making all in kconfig
make[2]: Entering directory '/home/mohamed/crosstool-ng-1.25.0/kconfig'
  LEX      lexer.lex.c
  YACC     parser.tab.c
parser.y:103.1-11: warning: POSIX Yacc does not support %destructor [-Wyacc]
  103 | %destructor {
      | ^~~~~~~~~~~
/usr/bin/make  all-am
make[3]: Entering directory '/home/mohamed/crosstool-ng-1.25.0/kconfig'
  CC       conf.o
  CC       confdata.o
  CC       expr.o
  CC       symbol.o
  CC       preprocess.o
  CC       util.o
  CC       parser.tab.o
  CC       lexer.lex.o
  CCLD     conf
  CC       nconf-nconf.o
  CC       nconf-nconf.gui.o
  CC       nconf-confdata.o
  CC       nconf-expr.o
  CC       nconf-symbol.o
  CC       nconf-preprocess.o
  CC       nconf-util.o
  CC       nconf-parser.tab.o
  CC       nconf-lexer.lex.o
  CCLD     nconf
  CC       mconf.o
  CC       lxdialog/checklist.o
  CC       lxdialog/inputbox.o
  CC       lxdialog/menubox.o
  CC       lxdialog/textbox.o
  CC       lxdialog/util.o
  CC       lxdialog/yesno.o
  CCLD     mconf
make[3]: Leaving directory '/home/mohamed/crosstool-ng-1.25.0/kconfig'
make[2]: Leaving directory '/home/mohamed/crosstool-ng-1.25.0/kconfig'
make[2]: Entering directory '/home/mohamed/crosstool-ng-1.25.0'
  GEN      ct-ng
  GEN      bash-completion/ct-ng
  GEN      docs/ct-ng.1
make[2]: Leaving directory '/home/mohamed/crosstool-ng-1.25.0'
make[1]: Leaving directory '/home/mohamed/crosstool-ng-1.25.0'

**Comment by MOHAMED19OS** [comment]
@cpackham 
i' test this change in make for file complete  successful
